File: BFF/src/main/java/fr/teama/bff/entities/Table.java
package fr.teama.bff.entities;

public class Table {

    private Long number;

    private boolean taken;

    private KioskOrder order;

    public Table(Long number) {
        this.number = number;
        this.taken = false;
    }

    public Table(Long number, boolean taken) {
        this.number = number;
        this.taken = taken;
    }

    public KioskOrder getOrder(){
        return this.order;
    }

    public boolean isTaken() {
        return taken;
    }

    public void takeTable(KioskOrder order){
            boolean taken = true;
            this.order = order;
        }

        public void freeTable(){
            boolean taken = false;
            this.order = null;
        }

    public Long getNumber() {
        return number;
    }
}

File: front-end-resto/src/app/models/Table.ts

class Table {
  private number: number;
  private taken: boolean;

  constructor(number: number, taken: boolean) {
    this.number = number;
    this.taken = taken;
  }

  getNumber(): number {
    return this.number;
  }

  setNumber(number: number): void {
    this.number = number;
  }

  isTaken(): boolean {
    return this.taken;
  }

  setTaken(taken: boolean): void {
    this.taken = taken;
  }

  equals(other: Table): boolean {
    return this.taken === other.isTaken() && this.number === other.getNumber();
  }
}

export default Table;
